spec/controllers/analytics/facilities_controller_spec.rb

```diff
@@ -37,18 +37,55 @@
   describe '#show' do
     render_views
 
-    it 'returns relevant analytics keys per facility' do
-      get :show, params: { id: facility.id }
+    context 'dashboard analytics' do
+      it 'returns relevant analytics keys per facility' do
+        get :show, params: { id: facility.id }
 
-      expect(response.status).to eq(200)
-      expect(assigns(:analytics)[user.id].keys).to match_array([:follow_up_patients_by_month,
-                                                                :registered_patients_by_month,
-                                                                :total_registered_patients])
+        expect(response.status).to eq(200)
+        expect(assigns(:analytics).dig(:dashboard, user.id).keys).to match_array([:follow_up_patients_by_month,
+                                                                                  :registered_patients_by_month,
+                                                                                  :total_registered_patients])
+      end
     end
 
     it 'renders the analytics table view' do
       get :show, params: { id: facility.id }
       expect(response).to render_template(partial: 'shared/_analytics_table')
     end
+
+    context 'analytics caching for facilities' do
+      before do
+        Rails.cache.clear
+      end
+
+      let(:cohort_date1) { (Date.today - (0 * 3).months).beginning_of_quarter }
+      let(:cohort_date2) { (Date.today - (1 * 3).months).beginning_of_quarter }
+      let(:cohort_date3) { (Date.today - (2 * 3).months).beginning_of_quarter }
+
+      it 'caches the facility correctly' do
+        today = Date.today.strftime("%Y-%m-%d")
```

Comment:
Instead of repeating `Date.today`, do this:

```ruby
let(:today) { Date.today }
```

Then use `today` in your other variables and specs. This prevents the race condition where a test flaps because it runs at midnight and spans two days.

Reply:
@timcheadle Yes, that was a silly oversight on my part! Updated the specs.